src/llmcompressor/modifiers/utils/helpers.py

```diff
@@ -72,10 +76,16 @@ def _valid_tensor_group_quant(layer_list: List[Linear]):
                 )
             ).reshape([1])
 
-            update_parameter_data(submodule.q_proj, global_scale, "weight_global_scale")
+        with align_module_device(submodule.k_proj):
```

Comment:
You don't need to onload the module before updating its parameters. Simply call `update_parameter_data` directly